### PR TITLE
Update dl to ol in ProfileInfoTable

### PR DESCRIPTION
**Pull request by sanlouise**
[TICKET](https://github.com/department-of-veterans-affairs/va.gov-team/issues/9785)

## Description
We needed to update the `dl` to an `ol` per accessibility's request. Also, this PR ensures it is now possible to leave out the `title` so a row can consist of just text.

## Testing done
Updated unit tests, works locally. The tables look identical to how they looked prior to the implementation.

## Screenshots
image

## Acceptance criteria
- [x] Update `dl` to `ol`
- [x] Make it so the table also accepts text only, leaving out title 

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Review by erikphansen — APPROVED** [review]
Looks good! I am wondering where in our CSS we hide the numbers for the `ol`.

File: src/applications/personalization/profile-2/components/ProfileInfoTable.jsx [hunk 1]
```diff
@@ -69,26 +69,26 @@ const ProfileInfoTable = ({
...
   };
 
-  const dlAttributes = list ? { role: 'list' } : null;
-  const dtAttributes = list ? { role: 'listitem' } : null;
-
   return (
     <section className={classes.table}>
       {title && <h3 className={classes.title}>{title}</h3>}
-      <dl className="vads-u-margin--0" {...dlAttributes}>
+      {/* eslint-disable-next-line jsx-a11y/no-redundant-roles */}
```
[the remark this answers is not in the document]
Comment (bkjohnson) [reply src/applications/personalization/profile-2/components/ProfileInfoTable.jsx:75]: Why are we adding `role="list"` to an `<ol>` element?

File: src/applications/personalization/profile-2/components/ProfileInfoTable.jsx [hunk 2]
```diff
@@ -69,26 +69,26 @@ const ProfileInfoTable = ({
...
             element => (dataTransformer ? dataTransformer(element) : element),
           )
           .map((row, index) => (
-            <div key={index} className={classes.tableRow}>
-              <dt className={classes.tableRowTitle} {...dtAttributes}>
-                {row.title}
-              </dt>
-              <dd className={classes.tableRowData}>{row.value}</dd>
-            </div>
+            // eslint-disable-next-line jsx-a11y/no-redundant-roles
```
[the remark this answers is not in the document]
Comment (bkjohnson) [reply src/applications/personalization/profile-2/components/ProfileInfoTable.jsx:82]: And why are we adding `role="listitem"` to an `<li>` element?

**Review by bkjohnson — COMMENTED** [review]
@sanlouise I don't understand why you've disabled the eslint rules, because I don't see any styling that would remove the list semantics.  Can you explain?

**Comment by bkjohnson** [comment]
Nevermind - it looks like this is intentional.

**Pull request merged**